tests/unit_tests/scheduler/test_openpbs_driver.py

```diff
@@ -468,3 +468,61 @@ async def test_keep_qsub_output(
         assert " -o /dev/null -e /dev/null" in Path("captured_qsub_args").read_text(
             encoding="utf-8"
         )
+
+
+@pytest.mark.parametrize(
+    "job_return_code",
+    [
+        pytest.param(0, id="realization_finished_successfully"),
+        pytest.param(2, id="realization_finished_with_error"),
+    ],
+)
+async def test_that_kill_does_not_log_error_for_finished_realization(
+    job_return_code, caplog, capsys, tmp_path, monkeypatch
+):
+    bin_path = tmp_path / "bin"
+    bin_path.mkdir()
+    monkeypatch.setenv("PATH", f"{bin_path}:{os.environ['PATH']}")
+    qsub_path = bin_path / "qsub"
+    qsub_path.write_text("#!/bin/sh\necho '1'")
+    qsub_path.chmod(qsub_path.stat().st_mode | stat.S_IEXEC)
+    qstat_path = bin_path / "qstat"
+    qstat_path.write_text(
+        dedent(
+            """\
+        #!/usr/bin/env python3
+        import sys
+        import json
+
+        job = "1"
+
+        if "-Fjson" in sys.argv:
+            print(json.dumps({"Jobs": {"1": {"Job_Name": "1", "job_state": "F", "Exit_status": '"""
+            + str(job_return_code)
+            + """'}}}))
+        else:
+            QSTAT_HEADER = (
+            "Job id                         Name            User             Time Use S Queue\\n"
+            "-----------------------------  --------------- ---------------  -------- - ---------------\\n"
+        )
+            print(QSTAT_HEADER, end="")
+            user = "mock"
+            time = "00:00:00"
+            queue = "mocked"
+            state = "F"
+
+            print(
+                f"{job:30.30}  {job:15.15} {user:15.15}  {time:8.8} {state:1.1} {queue:5.5}"
+            )
+        """
+        )
+    )
+    qstat_path.chmod(qstat_path.stat().st_mode | stat.S_IEXEC)
```

Comment:
I have a gut feeling it should be possible to do `driver.poll = somethingmocked` to shorten this section.

Reply:
I am struggling to find a way to mock it without relying too much on the implementation detail. What about now?